code_samples/front/embed_content/src/Controller/RelationController.php

```diff
@@ -25,13 +25,13 @@ public function showContentAction(View $view, $locationId): View
         $location = $this->locationService->loadLocation($locationId);
         $contentInfo = $location->getContentInfo();
         $versionInfo = $this->contentService->loadVersionInfo($contentInfo);
-        $relations = $this->contentService->loadRelations($versionInfo);
+        $relationList = $this->contentService->loadRelationList($versionInfo);
 
         $items = [];
 
-        foreach ($relations as $relation) {
-            if (in_array($relation->getDestinationContentInfo()->getContentType()->identifier, $acceptedContentTypes)) {
-                $items[] = $this->contentService->loadContentByContentInfo($relation->getDestinationContentInfo());
+        foreach ($relationList as $relationListItem) {
+            if (in_array($relationListItem->getRelation()->getDestinationContentInfo()->getContentType()->identifier, $acceptedContentTypes)) {
```

Comment:
```suggestion
            if ($relationListItem->hasRelation() && in_array($relationListItem->getRelation()->getDestinationContentInfo()->getContentType()->identifier, $acceptedContentTypes)) {
```

Shouldn't we make sure that the relation is accessible here as well?

BTW: I will verify the config from PHPStan to make sure it catches issues like this

Reply:
@mnocon 
I make many tests to have some _UnauthorizedRelationListItem_ and didn't manage to without hacking.
I also noticed meanwhile that it can effectively list some hidden relation.

https://github.com/ibexa/core/blob/main/src/lib/Repository/ContentService.php#L2056
`$this->permissionResolver->canUser('content', 'read', $destinationContentInfo)` is true on an hidden object or an object without visible location. When they are children of an hidden location, they are simply not listed.
So, it seems to be messy. I could try harder. Maybe it could be kept for another time. Like when coming back to https://github.com/ibexa/documentation-developer/pull/2548/files, I could add there and on 4.6.